### PR TITLE
Generate API docs from ts/tsx files

### PR DESCRIPTION
**Pull request by galibhassan**
…gin.

Don't remember when i wrote this `...gin.` thing.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "build:css": "node-sass src/_BaseTable.scss ./styles.css --output-style expanded",
     "build": "npm run build:js && npm run build:es && npm run build:css",
     "build:typescript": "tsc",
+    "postinstall": "tsc",
```
Comment (kohakukun) [inline package.json]: should refer to the build and we should remove build:js

File: website/plugins/gatsby-transformer-react-docgen-typescript/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,291 @@
+# Change Log
```
Comment (kohakukun) [inline website/plugins/gatsby-transformer-react-docgen-typescript/CHANGELOG.md]: You can remove this file

File: website/plugins/gatsby-transformer-react-docgen-typescript/LICENSE [hunk 3]
```diff
@@ -0,0 +1,22 @@
+The MIT License (MIT)
```
Comment (kohakukun) [inline website/plugins/gatsby-transformer-react-docgen-typescript/LICENSE]: And this one

File: website/plugins/gatsby-transformer-react-docgen-typescript/README.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
+### [IMPORTANT]
```
Comment (kohakukun) [inline website/plugins/gatsby-transformer-react-docgen-typescript/README.md]: Remove information that doesnt apply to this script (e.g. the installation)

File: website/plugins/gatsby-transformer-react-docgen-typescript/gatsby-node.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+"use strict";
+
+exports.onCreateNode = require(`./on-node-create`).default;
+exports.setFieldsOnGraphQLNodeType = require(`./extend-node-type`).default;
```
Comment (kohakukun) [inline website/plugins/gatsby-transformer-react-docgen-typescript/gatsby-node.js]: I wonder if you can reuse the files that you copied without modifications

File: website/plugins/gatsby-transformer-react-docgen-typescript/index.js [hunk 6]
```diff
@@ -0,0 +1 @@
+// noop
```
Comment (kohakukun) [inline website/plugins/gatsby-transformer-react-docgen-typescript/index.js]: Remove this file

File: website/plugins/gatsby-transformer-react-docgen-typescript/package.json [hunk 7]
```diff
@@ -0,0 +1,51 @@
+{
```
Comment (kohakukun) [inline website/plugins/gatsby-transformer-react-docgen-typescript/package.json:1]: I would skim this package to the minimum since you are not publishing or anything, I would only leave the name to avoid confusions

File: website/siteConfig.js [hunk 8]
```diff
@@ -28,19 +28,51 @@ module.exports = {
   api: [
     {
       title: 'BaseTable',
-      path: '/api/basetable',
+      path: `/api/${`basetable`.toLowerCase()}`,
```
Comment (kohakukun) [inline website/siteConfig.js]: why this change? HTML is even agnostic to capitalization

File: website/siteConfig.js [hunk 9]
```diff
@@ -28,19 +28,51 @@ module.exports = {
...
+    {
+      title: 'TableCell',
+      path: `/api/${`TableCell`.toLowerCase()}`,
+    },
+    {
+      title: 'TableRow',
+      path: `/api/${`TableRow`.toLowerCase()}`,
+    },
+    {
+      title: 'TableHeader',
```
Comment (kohakukun) [inline website/siteConfig.js]: I think we should not expose more than what was exposed before

File: website/plugins/gatsby-transformer-react-docgen-typescript/gatsby-node.js (hunk 5 above)
Comment (galibhassan) [reply website/plugins/gatsby-transformer-react-docgen-typescript/gatsby-node.js]: still don't get it. reusing from where? 
You told me to use require,
The files copied without modification are needed in the `getsby-node.js` of the plugin. For example, `parse.js`. They are nowhere else (e.g. not in node_modules, etc). From where do I use require?

File: website/siteConfig.js (hunk 8 above)
Comment (galibhassan) [reply website/siteConfig.js]: Hmm. But if I change `basetable` to `BaseTable`, it shows a blank page. May be because of the following:
`./website/src/template/api.js`, line 24. 

````javascript
      const link = links.find(link => link.to === `/api/${name.toLowerCase()}`)
````
Where `links` have the form:

````javascript
[
  {
    "key": "BaseTable",
    "title": "BaseTable",
    "to": "/api/BaseTable"
  },
  {
    "key": "Column",
    "title": "Column",
    "to": "/api/column"
  },
  // ...
]
````

Comment (galibhassan) [reply website/siteConfig.js]: Anyway, If we don't need to expose any other than the previous four, we don't need it.

**Review by kohakukun — APPROVED** [review]
